File: Python/ship.py
from waterway import Waterway

class Ship(Waterway):

    def __init__(self, vehicle_id, vehicle_name, engine_type, capacity):
        super().__init__(vehicle_id, vehicle_name, engine_type)
        self.capacity = capacity

    def display_details(self):
        super().display_details()
        print("Cargo Capacity  :", self.capacity, "kg")

    def start_journey(self):
        print("Ship Journey Started Successfully...")

File: Java/Ship.java
class Ship extends Waterway
{
    private double cargoCapacity;

    Ship(int id, String name, String engineType, double cargoCapacity)
    {
        super(id, name, engineType);
        this.cargoCapacity = cargoCapacity;
    }

    public void displayDetails()
    {
        super.displayDetails();
        System.out.println("Cargo Capacity : " + cargoCapacity + " Tons");
    }

    @Override
    void startJourney()
    {
        System.out.println("Ship Journey Started Successfully...");
    }
}